lib/goal/container/k8s.ts

```diff
@@ -668,7 +668,7 @@ async function containerStarted(container: K8sContainer, attempts: number = 240)
  * @return Status of pod after container terminates
  */
 function containerWatch(container: K8sContainer, timeout: number): Promise<k8s.V1PodStatus> {
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
```

Comment:
I'll test to see. I got the impression that the `watch` function still returns before the "watching" is complete. Looking through the code, it seems it is async now because an underlying authentication mechanism it calls, or rather something it calls calls something that calls something that calls something, is async. The callbacks are still in play, https://github.com/kubernetes-client/javascript/blob/master/examples/typescript/watch/watch-example.ts .

Reply:
That does not work. Changing the code to return the promise returned by `watch`, and adding logging around the call to containerWatch and within the phase change callback, you get:

```
awaiting containerWatch
containerWatch complete
update:ADDED
update:MODIFIED
update:DELETED
```

which is not what we want. After adding similar logging to the existing function, you get:

```
awaiting containerWatch
update:ADDED
containerWatch complete
```

where the update comes before the containerWatch call returns.